identity/README.md

```diff
@@ -0,0 +1,299 @@
+  // dots (.), and alphanumerics between. This field is REQUIRED.
+  string name = 1;
+  // This field is REQUIRED. Value of this field is opaque to the CO.
+  string vendor_version = 2;
+  // This field is OPTIONAL. Values are opaque to the CO.
+  map<string, string> manifest = 3;
+}
+```
+
+#### GetIdentity Errors
+
+If the CSI-driver is unable to complete the GetIdentity call successfully, it
+MUST return a non-ok gRPC code in the gRPC status.
+
+### `GetCapabilities`
+
+The CSI-Addons Controller needs to detect the supported features for each
+CSI-driver. Features are described as capabilities, each with additional
+information on how the CSI-Addons Controller may invoke the operation.
+
+There might be operations that need to run on the CSI Controller (provisioner)
+for storage system access. Other operations may require running on a volume
+that is attached/staged on a node. By specifying the type of Service that the
+CSI-driver provides, the CSI-Addons Controller can make informed decisions on
+where to invoke the RPCs for an operation.
+
+```protobuf
+// GetCapabilitiesRequest is sent by the CSI-Addons Controller to detect the
+// features that a CSI-driver supports.
+message GetCapabilitiesRequest {
+  // Intentionally empty.
+}
+
+// GetCapabilitiesResponse is returned by the CSI-driver as a response to a
+// GetCapabilitiesRequest.
+message GetCapabilitiesResponse {
+  // All the capabilities that the controller service supports. This
+  // field is OPTIONAL.
+  repeated Capability capabilities = 1;
+}
+
+// Specifies one or more capabilities of the CSI-driver.
+message Capability {
+  // Service contains the type of CSI Service that the CSI-driver provides.
+  message Service {
+    // Type describes a CSI Service that CSI-drivers can support.
+    enum Type {
+      // UNKNOWN indicates that the CSI-driver does not neither provide the CSI
+      // ControllerService or CSI NodeService. The CSI-Addons Controller will
+      // most likely ignore the node providing this Identify Service.
+      UNKNOWN = 0;
+
+      // CONTROLLER_SERVICE indicates that the CSI-driver provides RPCs for the
+      // CSI ControllerService.
+      // The presence of this capability determines whether the CSI-Addons
+      // Controller can invoke RPCs that require access to the storage system,
+      // similar to the CSI Controller (provisioner).
+      CONTROLLER_SERVICE = 1;
+
+      // NODE_SERVICE indicates that the CSI-driver provides RPCs for the CSI
+      // NodeService.
+      // The presence of this capability determines whether the CSI-Addons
+      // Controller can invoke RPCs that require a volume to be staged/attached
+      // to the node.
+      NODE_SERVICE = 2;
+    }
+    // type contains the Type of CSI Service that the CSI-driver supports.
+    Type type = 1;
+  }
+
+  //message ReclaimSpace {
+  //
+  // example included for ReclaimSpace support
+  //
+  //}
```

Comment:
do we need this commented code? good to remove it or add when its required

Reply:
removed the example code, replaced with a note